From: Michal Kubecek <mkubecek@suse.cz>
Date: Mon, 27 Aug 2018 13:02:52 +0200
Subject: Revert "lib: Use existing define with polynomial"
Patch-mainline: Never, temporary hack
References: none

This reverts commit faa16bc404d72a5afb857c924c83a5f691f83386.

It's just a style cleanup but it breaks ppc64le build. Until the issue
is resolved in upstream, let's revert the commit.

Signed-off-by: Michal Kubecek <mkubecek@suse.cz>
---
 lib/decompress_bunzip2.c | 3 +--
 lib/xz/xz_crc32.c        | 3 +--
 2 files changed, 2 insertions(+), 4 deletions(-)

--- a/lib/decompress_bunzip2.c
+++ b/lib/decompress_bunzip2.c
@@ -51,7 +51,6 @@
 #endif /* STATIC */
 
 #include <linux/decompress/mm.h>
-#include <linux/crc32poly.h>
 
 #ifndef INT_MAX
 #define INT_MAX 0x7fffffff
@@ -655,7 +654,7 @@ static int INIT start_bunzip(struct bunzip_data **bdp, void *inbuf, long len,
 	for (i = 0; i < 256; i++) {
 		c = i << 24;
 		for (j = 8; j; j--)
-			c = c&0x80000000 ? (c << 1)^(CRC32_POLY_BE) : (c << 1);
+			c = c&0x80000000 ? (c << 1)^0x04c11db7 : (c << 1);
 		bd->crc32Table[i] = c;
 	}
 
--- a/lib/xz/xz_crc32.c
+++ b/lib/xz/xz_crc32.c
@@ -15,7 +15,6 @@
  * but they are bigger and use more memory for the lookup table.
  */
 
-#include <linux/crc32poly.h>
 #include "xz_private.h"
 
 /*
@@ -30,7 +29,7 @@ STATIC_RW_DATA uint32_t xz_crc32_table[256];
 
 XZ_EXTERN void xz_crc32_init(void)
 {
-	const uint32_t poly = CRC32_POLY_LE;
+	const uint32_t poly = 0xEDB88320;
 
 	uint32_t i;
 	uint32_t j;
